src/filepattern/cpp/util/util.hpp

```diff
@@ -141,6 +141,26 @@ namespace s {
         return file;
     }
 
+    /**
+    * @brief Get the basename of a filepath.
+    *
+    * @param file_path Filepath to find the basename of.
+    * @return string The basename of the filepath.
+    */
+    inline std::string getBaseName(const fs::path& filesystem_path) {
```

Comment:
where are we using this function? Can we use `fs::path::filename()` instead?

Reply:
This overloads the function "inline std::string getBaseName(const string& path)" so std::string and fs::path can be used interchangeably throughout the code without branching. For example, when java bindings are enabled the paths are stored as std::string. I think it would be best update the getBasename function to use fs::path::filename() but the function is needed to avoid branching throughout the code. Let me know if you think a different approach would be better

Reply:
Something like this may be?
```

    inline std::string getBaseName(const fs::path& filesystem_path) {
    return filesystem_path.filename().u8string();
    }
```